### PR TITLE
Mg/symfuncs

### PR DESCRIPTION
**Pull request by mgastegger**
All kinds of atom centered symmetry functions.

### CODE REVIEW

File: src/schnetpack/configs/model/representation/wacsf.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  _target_: schnetpack.representation.symfuncs.AngularSF
+  radial_basis:
+    _target_: schnetpack.nn.radial.GaussianRBFCentered
+    n_rbf: 5
+    cutoff: ${cutoff}
+  elemental_basis:
+    _target_: schnetpack.nn.symmetry_functions.WeightedElements
+  angular_basis:
+    _target_: schnetpack.nn.symmetry_functions.AngularBasis
+    zetas:
```
Comment (ktschuett) [inline src/schnetpack/configs/model/representation/wacsf.yaml:23]: perhaps in this case it is more readable to write `zetas: [1, 4, 8]`

File: src/schnetpack/data/loader.py [hunk 2]
```diff
@@ -21,7 +21,9 @@ def _atoms_collate_fn(batch):
         dict[str->torch.Tensor]: mini-batch of atomistic systems
     """
     elem = batch[0]
-    idx_keys = {structure.idx_i, structure.idx_j}
+    idx_keys = {structure.idx_i, structure.idx_j, structure.idx_i_triples}
```
Comment (ktschuett) [inline src/schnetpack/data/loader.py:24]: i think the idx_j_triples are now treated a second time in ll 28-29, right?

File: src/schnetpack/model/base.py [hunk 3]
```diff
@@ -39,11 +39,11 @@ def __init__(
         self._postproc_cfg = postprocess or []
         self.inference_mode = False
 
-        self.build_model()
+        self.build_model(datamodule)
         self.build_postprocess(datamodule)
 
     @abstractmethod
-    def build_model(self):
+    def build_model(self, datamodule):
```
Comment (ktschuett) [inline src/schnetpack/model/base.py:46]: type hint, see below

File: src/schnetpack/model/pes.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ class PESModel(AtomisticModel):
     AtomisticModel for potential energy surfaces
     """
 
-    def build_model(self):
+    def build_model(self, datamodule):
```
Comment (ktschuett) [inline src/schnetpack/model/pes.py:21]: type hint

File: src/schnetpack/model/single_property.py [hunk 5]
```diff
@@ -15,9 +15,7 @@ class SinglePropertyModel(AtomisticModel):
     AtomisticModel for models that predict single chemical properties, e.g. for QM9 benchmarks.
     """
 
-    def build_model(
-        self,
-    ):
+    def build_model(self, datamodule):
```
Comment (ktschuett) [inline src/schnetpack/model/single_property.py:18]: type hint

File: src/schnetpack/configs/model/pes_sf.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    elements:
+      - 1
+      - 6
+      - 8
+    n_hidden:
+      - 100
+      - 100
+
+  normalization:
+    # TODO: trainable: True
```
[the remark this answers is not in the document]
Comment (mgastegger) [reply src/schnetpack/configs/model/pes_sf.yaml]: Changed.

File: src/schnetpack/configs/model/representation/wacsf.yaml (hunk 1 above)
Comment (mgastegger) [reply src/schnetpack/configs/model/representation/wacsf.yaml]: Changed.

File: src/schnetpack/data/loader.py (hunk 2 above)
Comment (mgastegger) [reply src/schnetpack/data/loader.py:24]: Fixed.

File: src/schnetpack/model/pes.py (hunk 4 above)
Comment (mgastegger) [reply src/schnetpack/model/pes.py]: Added.

File: src/schnetpack/model/base.py (hunk 3 above)
Comment (mgastegger) [reply src/schnetpack/model/base.py]: Added.

File: src/schnetpack/model/single_property.py (hunk 5 above)
Comment (mgastegger) [reply src/schnetpack/model/single_property.py]: Added.

**Review by ktschuett — APPROVED** [review]

**Pull request merged**